### PR TITLE
forcing the eventType to be PageAction. If you use the directives of …

### PR DESCRIPTION
**Pull request by danenone**
…angulartics on href the eventType will be click. New Relic is not accept that as valid.

### CODE REVIEW

**Review by gampleman — APPROVED** [review]

File: lib/angulartics-newrelic-insights.js [hunk 1]
```diff
@@ -16,6 +16,7 @@
     .config(['$analyticsProvider', function($analyticsProvider) {
       angulartics.waitForVendorApi('newrelic', 100, function(newrelic) {
         $analyticsProvider.registerEventTrack(function(action, properties) {
+          properties.eventType = 'PageAction';
```
Comment (gampleman) [inline lib/angulartics-newrelic-insights.js]: On further reflection, mutating an object that we don't own seems like a bad idea.

**Review by gampleman — CHANGES_REQUESTED** [review]

File: lib/angulartics-newrelic-insights.js (hunk 1 above)
Comment (danenone) [reply lib/angulartics-newrelic-insights.js]: I don't disagree with you at all. Here is the issue. I have a element `<a class="btn btn-sm btn-warning"
                                   analytics-on
                                   analytics-event="itemAddedToCart-productView"
                                   analytics-category="cart"
                                   analytics-label={{productCtrl.product.id}}
                                   ng-click="productCtrl.adjustCartQuantity(productCtrl.product)"
                                   ng-if="productCtrl.showAddToCartButton()">`

this sends an eventType of "click" to new relic. New Relic responses with a 400 Bad Request. Do you have another suggestion for me to look at? or would it be better for me to change a few values ie if eventType === 'click' then eventType = 'PageAction' so we are deliberate in how the object is modified?

Comment (gampleman) [reply lib/angulartics-newrelic-insights.js]: I think the solution would be to clone the object before it gets mutated, so something like:

```javascript
$analyticsProvider.registerEventTrack(function(action, properties) {
   var propertiesCopy = angular.copy(properties);
   propertiesCopy.eventType = 'PageAction';
   newrelic.addPageAction(action, propertiesCopy);
});
```

Also if you could add a link as a comment to some documentation that explains why this is necessary, that would be great.

Comment (danenone) [reply lib/angulartics-newrelic-insights.js]: I will make the changes. I am contacting new relic to see if I can get some official documentation about it.

**Comment by danenone** [comment]
I have not heard back on the documentation from our new relic rep yet.

**Comment by danenone** [comment]
@gampleman I think I have made all the changes you requested.

**Review by gampleman — APPROVED** [review]

**Pull request merged**